docs/security.md

```diff
@@ -0,0 +1,121 @@
+# Security
+
+The OpenTelemetry Collector defaults to operating in a secure manner, but is
+configuration driven. This document captures important security aspects and
+considerations for the Collector.
+
+## Configuration
+
+The Collector does not contain an embedded or default configuration and MUST
+NOT start without a configuration file being specified. The configuration file
+passed to the Collector MUST be validated prior to be loaded. If an invalid
+configuration is detected, the Collector MUST fail to start as a protective
+mechanism.
+
+The configuration drives the Collector's behavior and care should be taken to
+ensure the configuration only enables the minimum set of capabilities and as
+such exposes the minimum set of required ports. In addition, any incoming or
+outgoing communication SHOULD leverage TLS and/or authentication.
+
+The Collector keeps the configuration in memory, but where the configuration is
+loaded from at start time depends on the packaging used. For example, in
+Kubernetes secrets and configmaps CAN be leveraged. In comparison, the Docker
+image embeds the configuration in the container where is it not stored in an
+encrypted manner by default.
+
+The configuration MAY contain sensitive information including:
+
+- Authentication information such as API tokens
+- TLS certificates including private keys
+
+Sensitive information SHOULD be stored securely such as on an encrypted
+filesystem or secret store. Environment variable CAN be used to handle
+sensitive and non-sensitive data as the Collector MUST support environment
+variable expansion.
```

Comment:
If an end-user is reading this to learn how to set up a certain component I think they would benefit from a link to [this docs section](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables) to see how this is done in practice. Not sure if a link is appropriate here though, this sounds more like a spec for Collector developers.